mmv1/products/vmwareengine/Cluster.yaml

```diff
@@ -0,0 +1,145 @@
+base_url: "{{parent}}/clusters"
+create_url: "{{parent}}/clusters?clusterId={{name}}"
+self_link: "{{parent}}/clusters/{{name}}"
+update_mask: true
+update_verb: :PATCH
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: "https://cloud.google.com/vmware-engine/docs/reference/rest/v1/projects.locations.privateClouds.clusters"
+description: |
+  A cluster in a private cloud.
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 5000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 160
+      update_minutes: 160
+      delete_minutes: 120
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+  include_project: true
+
+import_format: ["{{%parent}}/clusters/{{name}}"]
+id_format: "{{parent}}/clusters/{{name}}"
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vmware_engine_cluster_basic"
+    min_version: beta
+    primary_resource_id: "vmw-engine-ext-cluster"
+    skip_test: true   # update tests will take care of create and update. PC and cluster creation is expensive and node reservation is required.
+    vars:
+      name: "ext-cluster"
+      private_cloud_id: "sample-pc"
+      network_id: "pc-nw"
+      management_cluster_id: "sample-mgmt-cluster"
+    test_env_vars:
+      region: :REGION
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vmware_engine_cluster_full"
+    min_version: beta
+    primary_resource_id: "vmw-ext-cluster"
+    skip_test: true   # update tests will take care of create and update. PC and cluster creation is expensive and node reservation is required.
+    vars:
+      name: "ext-cluster"
+      private_cloud_id: "sample-pc"
+      network_id: "pc-nw"
+      management_cluster_id: "sample-mgmt-cluster"
+    test_env_vars:
+      region: :REGION
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "parent"
+    immutable: true
+    required: true
+    url_param_only: true
+    description: |
+      The resource name of the private cloud to create a new cluster in.
+      Resource names are schemeless URIs that follow the conventions in https://cloud.google.com/apis/design/resource_names.
+      For example: projects/my-project/locations/us-west1-a/privateClouds/my-cloud
+  - !ruby/object:Api::Type::String
+    name: "name"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The ID of the Cluster.
+
+properties:
+  - !ruby/object:Api::Type::Boolean
+    name: management
+    output: true
+    description: |
+      True if the cluster is a management cluster; false otherwise.
+      There can only be one management cluster in a private cloud and it has to be the first one.
```

Comment:
Actually PrivateCloud management cluster update/read or import is handled.
This property in general holds true, I thought of deleting this property altogether.. But some can still import a management cluster independently also, I think it might be useful that time.